src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -175,6 +175,11 @@ protected async Task<AuthenticationResult> CacheTokenResponseAndCreateAuthentica
 
             ValidateAccountIdentifiers(fromServer);
 
+            if (ShouldNotCacheTokenResponse())
+            {
+                return GetAuthenticationResultFromTokenResponse(msalTokenResponse);
```

Comment:
This is not correct. We need to cache Account object - it is very imporant. It should be ok to not cache AT and ID token.

Reply:
Now caching id tokens  so that the account object can be created with tenant profiles in the tokenCacheInternal logic